### ISSUE TITLE
TCP Client does not recover from WiFi disconnection

### ISSUE THREAD

**Issue by sirockin** [biancode/node-modbus#5]
I have a simple implementation of a TCP client which initialises in autoconnect mode as follows:

```
{
    this.client = modbus.client.tcp.complete({
      'host': this.host,
      'port': this.port,
      'autoReconnect': true,
      'reconnectTimeout': 5000,
      'logEnabled': true,
      'logLabel': "ModbusTCP Client" + this.host + ":" + this.port
    }).connect();
```
It then polls a circular queue of read requests (to holding registers) using this code snippet:
```
this.client.readHoldingRegisters(curJob.address, curJob.size)
      .then(function (resp)
      {
        // Does some processing here
        ....

      })
      .catch(function (err)
      {
         // Process error here
      })
      .then(function ()
      {
        // Increments or restarts queue
        self.setupNextRead();
      })
  };
```
If I connect to a controller on another IP address then cut the connection (eg by unplugging the server's network cable or by turning off my WiFi card), the catch handler is called with err.err === "timeout" (as expected). However, when the connection is restored, the client does not recover and continues to call the catch handler with the same timeout error for all subsequent requests.

This issue does not happen if I disconnect by switching off the modbus server device.

I have fixed in my code by calling .close() then .connect() in the catch handler. But surely this shouldn't be necessary?

Other info:
Running on MacOS 10.13.2, node-modbus version 4.0.1

**Comment by sirockin** [comment]
You can reproduce this easily by doing the following:
1. Run `examples\tcp\AutoReconnect.js` (corrected for issue #6) to connect to modbus server on a different IP
2. Disconnect then reconnect the ethernet connection

The program output seems to indicate that on reconnection, something is being left in the internal queue, as both success and failure messages are being shown
```
INFO : ModbusTCPClient log level: info
INFO : ModbusTCPClient log level: info
client connected.
Starting request...
Success 1 Errors 0 Reconnect 0
Request finished successfull.
Starting request...
Success 2 Errors 0 Reconnect 0
Request finished successfull.
Starting request...
Success 3 Errors 0 Reconnect 0
Request finished successfull.
Starting request...
Success 4 Errors 0 Reconnect 0
Request finished successfull.
Starting request...
Success 5 Errors 0 Reconnect 0
Request finished successfull.
Starting request...
ERROR : ModbusTCPClient Request timed out after 2 sec
{ err: 'timeout' }
Success 5 Errors 1 Reconnect 0
Request finished UNsuccessfull.
Client closed, stopping interval.
client connected.
Starting request...
ERROR : ModbusTCPClient Request timed out after 2 sec
{ err: 'timeout' }
Success 5 Errors 2 Reconnect 1
Request finished UNsuccessfull.
Client closed, stopping interval.
client connected.
Starting request...
Success 6 Errors 2 Reconnect 2
Request finished successfull.
Starting request...
ERROR : ModbusTCPClient Request timed out after 2 sec
{ err: 'timeout' }
Success 6 Errors 3 Reconnect 2
Request finished UNsuccessfull.
Client closed, stopping interval.
client connected.
Starting request...
Success 7 Errors 3 Reconnect 3
Request finished successfull.
Starting request...
ERROR : ModbusTCPClient Request timed out after 2 sec
{ err: 'timeout' }
Success 7 Errors 4 Reconnect 3
Request finished UNsuccessfull.
Client closed, stopping interval.
client connected.
Starting request...
Success 8 Errors 4 Reconnect 4
Request finished successfull.
Starting request...
ERROR : ModbusTCPClient Request timed out after 2 sec
{ err: 'timeout' }
Success 8 Errors 5 Reconnect 4
Request finished UNsuccessfull.
Client closed, stopping interval.
client connected.
Starting request...
Success 9 Errors 5 Reconnect 5
Request finished successfull.
Starting request...
ERROR : ModbusTCPClient Request timed out after 2 sec
{ err: 'timeout' }
Success 9 Errors 6 Reconnect 5
Request finished UNsuccessfull.
```

**Comment by biancode** [comment]
please, check v4.2.0
